src/Sylius/Behat/Context/Setup/PaymentMethodContext.php

```diff
@@ -0,0 +1,103 @@
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Context\Setup;
+
+use Behat\Behat\Context\Context;
+use Sylius\Bundle\CoreBundle\Test\Services\PaymentMethodNameToGatewayConverterInterface;
+use Sylius\Component\Core\Test\Services\SharedStorageInterface;
+use Sylius\Component\Payment\Model\PaymentMethodInterface;
+use Sylius\Component\Payment\Repository\PaymentMethodRepositoryInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+final class PaymentMethodContext implements Context
+{
+    /**
+     * @var SharedStorageInterface
+     */
+    private $sharedStorage;
+
+    /**
+     * @var FactoryInterface
+     */
+    private $paymentMethodFactory;
+
+    /**
+     * @var PaymentMethodNameToGatewayConverterInterface
+     */
+    private $paymentMethodNameToGatewayConverter;
+
+    /**
+     * @var PaymentMethodRepositoryInterface
+     */
+    private $paymentMethodRepository;
+
+    /**
+     * @param SharedStorageInterface $sharedStorage
+     * @param FactoryInterface $paymentMethodFactory
+     * @param PaymentMethodNameToGatewayConverterInterface $paymentMethodNameToGatewayConverter
+     * @param PaymentMethodRepositoryInterface $paymentMethodRepository
+     */
+    public function __construct(
+        SharedStorageInterface $sharedStorage,
+        FactoryInterface $paymentMethodFactory,
+        PaymentMethodNameToGatewayConverterInterface $paymentMethodNameToGatewayConverter,
+        PaymentMethodRepositoryInterface $paymentMethodRepository
+    ) {
+        $this->sharedStorage = $sharedStorage;
+        $this->paymentMethodFactory = $paymentMethodFactory;
+        $this->paymentMethodNameToGatewayConverter = $paymentMethodNameToGatewayConverter;
+        $this->paymentMethodRepository = $paymentMethodRepository;
+    }
+
+    /**
+     * @Given the store has a payment method :paymentMethodName with a code :paymentMethodCode
+     */
+    public function theStoreHasAPaymentMethodWithACode($paymentMethodName, $paymentMethodCode)
+    {
+        $paymentMethod = $this->createPaymentMethodFromNameAndCode($paymentMethodName, $paymentMethodCode);
+
+        $this->sharedStorage->set('payment_method', $paymentMethod);
+        $this->paymentMethodRepository->add($paymentMethod);
+    }
+
+    /**
+     * @Given /^the store has a payment method "([^"]*)" disabled$/
+     */
+    public function theStoreHasAPaymentMethodDisabled($paymentMethodName)
+    {
+        $paymentMethod = $this->createPaymentMethodFromNameAndCode($paymentMethodName);
+        $paymentMethod->disable();
+
+        $this->sharedStorage->set('payment_method', $paymentMethod);
+        $this->paymentMethodRepository->add($paymentMethod);
+    }
+
+    /**
+     * @param string $name
+     * @param string $code
+     *
+     * @return PaymentMethodInterface
+     */
+    private function createPaymentMethodFromNameAndCode($name, $code = 'PM')
```

Comment:
:+1: Moreover, what do you think about adding here another optional parameter ``$enabled``? Therefore, we could determine it method enabled/disabled in this function, and extract common logic from two functions above into one private method.